src/member-api.js

```diff
@@ -30,6 +30,7 @@ import { ROLES, isRoleIdForNewInvite } from './roles.js'
  * @prop {import('./roles.js').Role} role
  * @prop {import('@mapeo/schema').DeviceInfo['name']} [name]
  * @prop {import('@mapeo/schema').DeviceInfo['deviceType']} [deviceType]
+ * @prop {import('@mapeo/schema').DeviceInfo['createdAt']} [joinedAt]
```

Comment:
Is there any way we can make this required? It seems like it should always be defined, at least theoretically. Might require a larger change.

Reply:
yeah, was going through that. The easiest way I'm thinking of is turning the result ([here](https://github.com/digidem/mapeo-core-next/blob/8784111e312eb46de46adbd7a7725220c6133b1d/src/member-api.js#L245)) into a `SetOptional<MemberInfo,'joinedAt'>` when declared. Is there a better type wizardry for this?

Reply:
Mhm, I think is basically the same making it an optional field. I always struggle with ts when declaring an object of a type where some members are added some lines later. Is there a way to signal this to the type system?? (basically, 'I know `joinedAt` is missing, just look some lines later!)

Reply:
In this case I think the issue is that calling `this.#coreOwnership.getCoreId` and `this.#dataTypes.deviceInfo.getByDocId` can throw. So if that happens then `joinedAt` would be `undefined` (as with the other optionals of `MemberInfo`)

Reply:
Ah, I guess if everything else can be undefined then this is okay too.